### ISSUE TITLE
bot+app: Issue and Accept challenges

### ISSUE THREAD

**Issue by rsodre** [underware-gg/pistols-discord-bot#8]
Depends on #1, #5, #7

## Tasks

- [ ] Store duelist account data for Discord Users
  - [ ] Adapt the `/social-app` to store a new field: `account_data`
  - [ ] It holds a simple json string, exported from the [Pistols](https://pistols.lootunder.world/gate) client
- [ ] Add Starknet connectors to `/discord-bot`
  - [ ] TBD
- [ ] Create new command: `issue_challenge`
  - [ ] Input: `challenged_address`, containing the challenged duelist OR Discord user Id
  - [ ] Input: `wager_value` (optional)
  - [ ] Input: `message` (optional)
  - [ ] Fetch the user's account data from supabase.
  - [ ] Call the `create_challenge()` method on the `[actions](pistols::actions)` system
- [ ] Create new command: `accept_challenge`
  - [ ] Input: `duel_id`
  - [ ] Input: `accept`, a boolean as True by default
  - [ ] Fetch the user's account data from supabase.
  - [ ] Call the `reply_challenge()` method on the `pistols::actions` system

The [actions interface](https://github.com/underware-gg/pistols/blob/main/dojo/src/systems/actions.cairo) on Pistols:
```rust
    fn create_challenge(self: @TContractState,
        challenged: ContractAddress,
        message: felt252,
        wager_coin: u8,
        wager_value: u256,
        expire_seconds: u64,
    ) -> u128;
    fn reply_challenge(self: @TContractState,
        duel_id: u128,
        accepted: bool,
    ) -> ChallengeState;
```

## Notes

* If the commands are successful, the bot will automatically display notifications
* If they fail, the bot must forward the reason to the player on Discord as a private message
* We're working on the details, meanwhile this issue is **ON HOLD**
* Let's also consider using [Argent Session Keys](https://docs.argent.xyz/starknet/aa-use-cases/session-keys-early-access-feature)?
